src/Desktop/UIAutomation/CustomObjects/Registrar.cs

```diff
@@ -59,6 +59,29 @@ public void RegisterCustomProperty(CustomProperty prop)
         public Dictionary<int, CustomProperty> GetCustomPropertyRegistrations() { return new Dictionary<int, CustomProperty>(_idToCustomPropertyMap); }
 #pragma warning restore CA1024 // Use properties where appropriate: this is a copy of the object, not the object itself
 
+        /// <summary>
+        /// Notifies ConverterRegistrationAction about all of the custom property registrations in
+        // the passed-in dictionary. This is used, for instance, when switching between file and
+        // live mode in Accessibility Insights for Windows.
+        /// </summary>
+        /// <remarks>This function assumes that when ConverterRegistrationAction is called multiple times, the last call wins.</remarks>
+        public void MergeCustomPropertyRegistrations(IReadOnlyDictionary<int, CustomProperty> registrations)
+        {
+            if (registrations== null) throw new ArgumentNullException(nameof(registrations));
+            foreach (KeyValuePair<int, CustomProperty> entry in registrations)
+            {
+                _converterRegistrationAction(entry.Key, CreateTypeConverter(entry.Value));
+            }
+        }
+
+        /// <summary>
+        // Re-notifies converterRegistrationAction of the custom properties registered with UIA, in
+        // effect restoring IDs registered in live mode to the state before any calls to
+        // MergeCustomPropertyRegistrations.
+        // </summary>
```

Comment:
These should all be triple-slash comments
```suggestion
        /// Re-notifies converterRegistrationAction of the custom properties registered with UIA, in
        /// effect restoring IDs registered in live mode to the state before any calls to
        /// MergeCustomPropertyRegistrations.
        /// </summary>
```